lib/brain.js

```diff
@@ -1267,10 +1267,12 @@ Brain.prototype.smsFlowHandleReturnState = function smsFlowHandleReturnState ()
     return this._dispenseUpdate(this.tx)
   }
 
-  const amount = this.getUsersDailyVolume().add(this.lastRejectedBillFiat)
+  const amount = this.complianceDailyVolumeAmount()
   const nonCompliant = this.nonCompliantTiers(amount)
 
-  const otherTiers = _.isNil(this.failedCompliance) && !_.isEmpty(nonCompliant)
+  const otherTiers = _.isNil(this.failedCompliance) &&
+    !_.isEmpty(nonCompliant) &&
+    _.head(nonCompliant) !== 'hardLimit'
```

Comment:
I moved to this check instead to skip the `return complianceTiers.run(_.head(nonCompliant))` with _hardLimit_ that would crash

Reply:
But if there's a hardLimit, won't it just skip compliance? Where will it stop the tx if the user has exceeded hardLimit?

Reply:
This currently throw [here](https://github.com/lamassu/lamassu-machine/blob/v3/lib/compliance-tiers.js#L40)

Reply:
Right but that's never supposed to happen. Hitting the hardLimit is supposed to happen and shouldn't throw an error, it should make the user complete the transaction and not allow any more bills.